darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -81,17 +83,28 @@ def get_encoded_fig(mpl_fig: Any):
         encoded_fig = base64.b64encode(tmpfile.getvalue()).decode("utf-8")
         return encoded_fig
 
-    def generate_img(self):
+    def generate_fig(self):
         """
-        Generate the image using the figure data.
+        Generate a figure using the figure data.
         """
-        # generate the matplotlib figure using the figure's
+        # generate the figure using the figure's
         # function and function arguments
-        mpl_fig = self.fig_func(**self.fig_args)
-        # encode the matplotlib figure
-        encoded = self.get_encoded_fig(mpl_fig=mpl_fig)
-        # create the img string
-        self.img_str = f"<img src=data:image/png;base64,{encoded} alt={self.fig_title} width={self.fig_width}>"
+        fig = self.fig_func(**self.fig_args)
+        supported_mpl_fig_types = [matplotlib.figure.Figure, seaborn.axisgrid.JointGrid]
+        if type(fig) in supported_mpl_fig_types:
+            # encode the matplotlib figure
+            encoded = self.get_encoded_fig(mpl_fig=fig)
+            # create the img string
+            self.fig_html = f"<img src=data:image/png;base64,{encoded} alt={self.fig_title} width={self.fig_width}>"
+        elif isinstance(fig, pd.DataFrame):
+            # use built-in pandas utility to convert table to html
+            self.fig_html = fig.to_html(header=False, border=0)
```

Comment:
This line is currently not covered by the CI since there are no tests for it. This will have to be merged before we can add the "common access table" figure, but with that will come tests that will certainly cover this.